runtime/codert_vm/decomp.cpp

```diff
@@ -1348,31 +1348,43 @@ freeDecompilationRecord(J9VMThread * decompileThread, J9JITDecompilationInfo * i
 void  
 jitDataBreakpointAdded(J9VMThread * currentThread)
 {
+	J9JavaVM *vm = currentThread->javaVM;
+	J9JITConfig *jitConfig = vm->jitConfig;
+
 	/* We have exclusive */
 
 	Trc_Decomp_jitDataBreakpointAdded_Entry(currentThread);
 
-	++(currentThread->javaVM->jitConfig->dataBreakpointCount);
+	++(jitConfig->dataBreakpointCount);
 
-	/* Remove all breakpoints before resetting the methods */
+	bool inlineWatches = J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_JIT_INLINE_WATCHES);
+	if (!inlineWatches || !jitConfig->inlineFieldWatches) {
+		/* Remove all breakpoints before resetting the methods */
 
-	removeAllBreakpoints(currentThread);
+		removeAllBreakpoints(currentThread);
 
-	/* Toss the compilation queue */
+		/* Toss the compilation queue */
 
-	currentThread->javaVM->jitConfig->jitClassesRedefined(currentThread, 0, NULL);
+		currentThread->javaVM->jitConfig->jitClassesRedefined(currentThread, 0, NULL);
```

Comment:
`currentThread->javaVM->jitConfig` -> `jitConfig`

Reuse the local fetched above

Reply:
Fixed.